test/box-luatest/gh_7945_mvcc_assert_in_mvcc_test.lua

```diff
@@ -0,0 +1,125 @@
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all = function()
+    g.server = server:new{
+        alias   = 'default',
+        box_cfg = {memtx_use_mvcc_engine = true}
+    }
+    g.server:start()
+end
+
+g.after_all = function()
+    g.server:drop()
+end
+
+g.before_each(function()
+    g.server:exec(function()
+        box.schema.space.create('test')
+    end)
+end)
+
+g.after_each(function()
+    g.server:exec(function()
+        if box.space.test then
+            box.space.test:drop()
+        end
+    end)
+end)
+
+-- Test that checks that a read tracker that was set after gap write does not
+-- get in the way after the reader is committed.
+g.test_commit_after_gap_write = function()
+    g.server:exec(function()
+        box.space.test:create_index('primary', {type = 'tree'})
+
+        local t = require('luatest')
+        local txn_proxy = require('test.box.lua.txn_proxy')
```

Comment:
I like txn_proxy more because it is much simpler.. For example I has my own version of txn_proxy which emits lots of debug information, that allows me to debug failing tests.